opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1516,610 +2031,1764 @@ void CpGrid::addLgrsUpdateLeafView(const std::vector<std::array<int,3>>& cells_p
+                int refinedFace;
+                // Face might have vanished - Search its refined lgr-children faces in that case -
+                // last lgr where the face appears
+                if (elemLgrAndElemLgrFace_to_refinedLevelAndRefinedFace.count({elemLgr, preAdaptFace}) == 0) {
+                    // Get the index of the marked face where the refined face was born.
+                    const auto& markedFace = getParentFaceWhereNewRefinedFaceLiesOn(cells_per_dim_vec[shiftedLevel], preAdaptFace,
+                                                                                    markedElem_to_itsLgr[elemLgr], elemLgr);
+                    // Get the last LGR (marked element) where the marked face appeared.
+                    int lastLgrWhereMarkedFaceAppeared = faceInMarkedElemAndRefinedFaces[markedFace].back().first;
+                    const auto& lastLgrLevel = assignRefinedLevel[lastLgrWhereMarkedFaceAppeared];
+                    const auto& lastLgrLevelShifted = lastLgrLevel - preAdaptMaxLevel -1;
+                    const auto& lastAppearanceLgrEquivFace = replaceLgr1FaceIdxByLgr2FaceIdx(cells_per_dim_vec[shiftedLevel], preAdaptFace,
+                                                                                             markedElem_to_itsLgr[elemLgr],
+                                                                                             cells_per_dim_vec[lastLgrLevelShifted]);
+                    refinedFace = elemLgrAndElemLgrFace_to_refinedLevelAndRefinedFace[{lastLgrWhereMarkedFaceAppeared, lastAppearanceLgrEquivFace}][1];
+                    aux_refined_cell_to_face.push_back({refinedFace, face.orientation()});
+                }
+                // Face is stored in adapted_faces
+                else {
+                    assert( elemLgrAndElemLgrFace_to_refinedLevelAndRefinedFace.count({elemLgr, preAdaptFace}) == 1);
+                    refinedFace =  elemLgrAndElemLgrFace_to_refinedLevelAndRefinedFace[{elemLgr, preAdaptFace}][1];
+                    aux_refined_cell_to_face.push_back({refinedFace, face.orientation()});
+                }
+            } // end-cell_to_face
+            // Refined cell to face.
+            refined_cell_to_face_vec[shiftedLevel].appendRow(aux_refined_cell_to_face.begin(), aux_refined_cell_to_face.end());
+            // Get the cell geometry.
+            const auto& elemLgrGeom =  (*(elemLgrData->geometry_.geomVector(std::integral_constant<int,0>())))[elemLgrCellEntity];
+
+            // Create a pointer to the first element of "refined_cell_to_point" (required as the fourth argement to construct a Geometry<3,3> type object).
+            int* indices_storage_ptr = refined_cell_to_point_vec[shiftedLevel][cell].data();
+            refined_cells_vec[shiftedLevel][cell] = cpgrid::Geometry<3,3>(elemLgrGeom.center(), elemLgrGeom.volume(),
+                                                                          allLevelCorners, indices_storage_ptr);
```

Comment:
To me it seems that refined_cell_to_point_vec is a temporary container. It's entries are later copied to An instance of CpGridData.cell_to_point and the container will be deleted. Doesn't this mean that indices_storage_ptr will become a dangling pointer?

Reply:
Maybe we should/can later just reset this pointer in all the Geometries?
Or use the members on CpGridData for populating them directly?

Reply:
Indeed, thanks for catching it! It's fixed using your suggestion f implementing swap. See last commits :)